app/controllers/pair_requests_controller.rb

```diff
@@ -4,7 +4,9 @@ class PairRequestsController < ApplicationController
   # GET /pair_requests or /pair_requests.json
   def index
     @pair_request = PairRequest.new
-    @pair_requests = policy_scope(PairRequest).order_by_status.order_by_date
+    filter = params[:filter] == 'past' ? :past : :upcoming
+    @pair_requests = policy_scope(PairRequest).public_send(filter).order_by_status.order_by_date
```

Comment:
So I think this works great for the upcoming, but I wonder if it may be better for the user if when they are on the 'Past' tab, the records are ordered with the newest ones towards to top of the list? The way it is now, if you're on the past tab, the records furthest into the past are on at the top of the list, but these are the ones the user is less likely to be interacting with.